File: Akshat Jain/Day 7/ugly_number.py
class Solution:
    def isUgly(self, n: int) -> bool:
        
        if n == 0: return False

        sieve = (2, 3, 5)

        while n != 1:
            for f in sieve:
               if n % f == 0:
                   n = n // f
                   break
            else:
                return False

        return True

File: nishant_mishra/day_7/ugly_number.cpp
// https://leetcode.com/problems/ugly-number
class Solution {
public:
    bool isUgly(int n) {
        
        if(n == 1) return true;
        while(n >0){
            if(n == 1) return true;
            if(n % 2 == 0) n /= 2;
            else if(n % 3 == 0) n /= 3;
            else if(n % 5 == 0) n /= 5;   
            else break; 
        }
    
        return false;
    }
};
